text/0024-cron.md

```diff
@@ -0,0 +1,196 @@
+
+To keep datasets fresh, qri needs to automate the update process.  By _automate_ I mean: execute dataset updates as a background process, registered with the local operating system, according to some schedule I can specify. My computer should do the work of keeping datasets up to date for me, and should show me a log of what's been done for me, calling attention to significant changes.
+
+The magnitude of this shift to automation is noticeable. cron is _doing work for me_. It takes the process of dataset updating out of the categroy of chores, and opens up the possibliity that Qri can surprise me with new insights. 
+
+Adding cron starts to transition qri into territory best described as a _fog service_: a decentralized network of machines automatically producing local resources for a user that can aggregate to a broader network. Over time I'm hoping we can do more and more things under this "fog service" paradigm.
+
+Building a fog service requires special consideration to get right. This feature proposes running parts of Qri as a background process on a user's primary computing device. In this setting, we need to take great care to avoid interrupting a user's day-to-day computing experience, while also handling changing connectivity & resource availability. There is a very real chance that as a consequence of a cron job a user scheulded 6 months ago, Qri will be scheduled to start an intense script on a laptop with 5% battery left at a coffee shop while connected to restricted Wifi, while the user is nervously trying to get some dockerized python script to process a report that's due in an hour. We should aim to shape both the technical foundations and user experince of Qri to do the right thing in that context.
+
+This is a very high bar, but I think it's the standard we should work toward. Qri cron is both a big first step and a chance to iterate on meeting these requirements before including more parts of the Qri daemon in this "fog service" paradigm.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Here's a typical workflow for qri cron using datasets. This example shows working with the command line. As usual all of this would also be accessible over the local API. To start with, `qri cron` is a command with a number of subcommands. Ths first is `qri cron add`:
+
+```
+$ qri cron add me/dataset_name
+scheduled me/dataset_name
+updates every 2 days
+next update: Tues. April 11, 11:30:00
+
+WARNING: cron daemon is not currently running. start with:
+$ qri cron start
+to start the qri cron service. until then datasets will not auto-update.
+```
+
+In this case we've run `qri cron add` on a dataset that contains a valid `meta.accrualPeriodicity` field (in this case `R/P2D`). Qri used this value to infer a default update periodicity. If no `meta.accrualPeriodicity` were set, Qri would have returned an error:
```

Comment:
I find R/P hard to mentally process. Is this a standard syntax used somewhere I'm not familiar with? How about making it a flag like `--when 2D`.

Reply:
Yeah, it's [iso8601](https://en.wikipedia.org/wiki/ISO_8601#Repeating_intervals), which many metadata standards use. That being said, we don't _need_ to stick with it. Do you have other specs/standards we should consider?

Reply:
No, as long as there's a source of inspiration I'm good with it, I was unaware of this.